Porting/clean-commit

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/perl
+use Data::Dumper;
+use strict;
+use warnings;
+use Getopt::Long;
+use Pod::Usage;
+
+my $DEBUG=0;
+
+use constant {
+    NULL_SHA1 => ("0" x 40),
+    TAB => " " x 8,
```

Comment:
In most of the Perl code I've seen in the core distribution, 4 spaces is the equivalent of a tab.  I would say that it's about 75% 4-spaces, 23% 2-spaces, balance other.  So I think setting `TAB` to 8 is not consistent with our practice.  (I'm not claiming we're very consistent about this.)  I realize that in `pod/perlpod.pod` there is a section on "Verbatim Paragraph" that calls for 8-space tabs, but don't see any indication that that applies more generally.

Perhaps we could make this configurable with a command-line switch?

Reply:
> In most of the Perl code I've seen in the core distribution, 4 spaces is the equivalent of a tab.

No it not. The indent was 4 spaces, but a tab was always 8.

Reply:
> > In most of the Perl code I've seen in the core distribution, 4 spaces is the equivalent of a tab.
> 
> No it not. The indent was 4 spaces, but a tab was always 8.

I'm not going to get dogmatic about 8-space tab, but can you point us to where that's documented?